File: solutions/python/resistor-color-trio/2/resistor_color_trio.py
BANDS = ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"]

def label(colors: list) -> str:
    """Gives the label associated with a combination of three colours for a resistor
    
    :param colors: The list of colors from which colours will be translated into ohms.
    :return: int - The value of the colours combined into ohms.
    """

    value = (BANDS.index(colors[0]) * 10 + BANDS.index(colors[1])) * 10 ** BANDS.index(colors[2])

    if value >= 10 ** 9: return f"{int(value / 10 ** 9)} gigaohms"
    elif value >= 10 ** 6: return f"{int(value / 10 ** 6)} megaohms"
    elif value >= 10 ** 3: return f"{int(value / 10 ** 3)} kiloohms"        
    return f"{value} ohms"

File: solutions/php/resistor-color-trio/1/ResistorColorTrio.php
<?php

declare(strict_types=1);

class ResistorColorTrio
{
    public $band_colors = ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"];
    
    public function label(array $colors): string
    {
        $first_number = array_search($colors[0], $this->band_colors);
        $second_number = array_search($colors[1], $this->band_colors);
        $third_number = array_search($colors[2], $this->band_colors);

        $numero = (int) ($first_number . $second_number . str_repeat("0", $third_number));

        switch ($numero) {
            case ($numero < 1000):
                return $numero . " ohms";
            case ($numero < 1000000):
                return ($numero / 1000) . " kiloohms";
            case ($numero < 1000000000):
                return ($numero / 1000000) . " megaohms";
            case ($numero < 1000000000000):
                return ($numero / 1000000000) . " gigaohms";
            default:
                return $numero . " ohms";
        }
        
        return (string) $numero;
    }
}
